### PR TITLE
Add Flysystem pre-signed links

### PR DESCRIPTION
**Pull request by LucasCairns**
- Add patch to enable pre-signed links in Flysystem
- Update Composer JSON to apply patches
- Update Compose lock file
- Update Drupal to 8.9.2
- Update gitignore

### CODE REVIEW

**Comment by LucasCairns** [comment]
WIP - Presumably will also require additional configuration in Helm

File: Dockerfile [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-      curl \
-      git-core \
-      mediainfo \
-      unzip \
-    && rm -rf /var/lib/apt/lists/*
+  curl \
+  git-core \
+  mediainfo \
+  unzip \
+  && rm -rf /var/lib/apt/lists/*
```
Comment (spikeheap) [inline Dockerfile:9]: The extra indentation on L5-8 to visually distinguish the `&&`'d command in L9 feels more readable. Happy to see the indentation knocked back one tab, but could we leave that step?

File: Dockerfile [hunk 2]
```diff
@@ -17,22 +17,26 @@ RUN echo "date.timezone = Europe/London" > /usr/local/etc/php/conf.d/timezone_se
...
-  --no-scripts \
   --prefer-dist
 
 # Copy Project
-COPY flysystem/flysystem.* /var/www/html/modules/contrib/flysystem/
-COPY flysystem/FieldMigration.php /var/www/html/modules/contrib/flysystem/src/Form/
 COPY modules/custom modules/custom
 COPY sites/ sites/
 
+# Remove write persmissions for added security
```
Comment (spikeheap) [inline Dockerfile:36]: 👮 `persmissions` -> `permissions`

File: patches/flysystem_s3_8.x-1.0-beta1_enable_pre-signed_downloads.patch [hunk 3]
```diff
@@ -0,0 +1,426 @@
...
++    ]));
++    $client->getCommand('getObject', Argument::type('array'))->willReturn($this->prophesize(\Aws\Command::class)->reveal());
++    $request = new Psr7\Request('GET', 'https://s3.amazonaws.com/signed');
++    $client->createPresignedRequest(Argument::type('\Aws\Command'), '+10 seconds')->willReturn($request);
++    $client = $client->reveal();
++    $kill_switch = new KillSwitch();
++    $plugin = new S3($client, new Config($configuration), $this->renderer, $this->logger, $kill_switch);
++    $this->assertSame('https://s3.amazonaws.com/signed', $plugin->getExternalUrl('s3://foo.html'));
++    $this->assertEquals(ResponsePolicyInterface::DENY, $kill_switch->check(new Response(), new Request()));
++  }
```
Comment (spikeheap) [inline patches/flysystem_s3_8.x-1.0-beta1_enable_pre-signed_downloads.patch:379]: This patch introduces a bunch of tests 🎉. How can we run these tests (if we don't already)? Could we run this as part of the Docker build or CI?

File: Dockerfile (hunk 1 above)
Comment (LucasCairns) [reply Dockerfile:9]: Formatting fixed

File: Dockerfile (hunk 2 above)
Comment (LucasCairns) [reply Dockerfile:36]: Spelling corrected

File: patches/flysystem_s3_8.x-1.0-beta1_enable_pre-signed_downloads.patch (hunk 3 above)
Comment (LucasCairns) [reply patches/flysystem_s3_8.x-1.0-beta1_enable_pre-signed_downloads.patch:379]: Will spin up a card for this

**Pull request merged**